homeassistant/components/sensor/nzbget.py

```diff
@@ -1,48 +1,66 @@
 """
-Support for monitoring NZBGet nzb client.
+Support for monitoring NZBGet NZB client.
 
-Uses NZBGet's JSON-RPC API to query for monitored variables.
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.nzbget/
 """
 import logging
 from datetime import timedelta
+
 import requests
+import voluptuous as vol
 
+from homeassistant.const import (CONF_HOST, CONF_PASSWORD, CONF_USERNAME,
+                                 CONF_PLATFORM, CONF_NAME, CONF_PORT,
+                                 CONTENT_TYPE_JSON, CONF_MONITORED_VARIABLES)
 from homeassistant.helpers.entity import Entity
 from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+DEFAULT_NAME = 'NZBGet'
+DEFAULT_PORT = 6789
 
-REQUIREMENTS = []
 SENSOR_TYPES = {
-    "ArticleCacheMB": ("Article Cache", "MB"),
-    "AverageDownloadRate": ("Average Speed", "MB/s"),
-    "DownloadRate": ("Speed", "MB/s"),
-    "DownloadPaused": ("Download Paused", None),
-    "FreeDiskSpaceMB": ("Disk Free", "MB"),
-    "PostPaused": ("Post Processing Paused", None),
-    "RemainingSizeMB": ("Queue Size", "MB"),
+    'article_cache': ('Article Cache', 'MB'),
+    'average_download_rate': ('Average Speed', 'MB/s'),
+    'download_rate': ('Speed', 'MB/s'),
+    'download_paused': ('Download Paused', None),
+    'free_disk_space': ('Disk Free', 'MB'),
+    'post_paused': ('Post Processing Paused', None),
+    'remaining_size': ('Queue Size', 'MB'),
 }
+
 DEFAULT_TYPES = [
-    "DownloadRate",
-    "DownloadPaused",
-    "RemainingSizeMB",
+    'download_rate',
+    'download_paused',
+    'remaining_size',
 ]
 
+PLATFORM_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): 'nzbget',
```

Comment:
Oh, this one will actually work because it doesn't extend the Sensor default schema (which it should).

```python
from homeassistant.components.sensor import PLATFORM_SCHEMA

PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({…})
```

When extending, do not include the CONF_PLATFORM pointer to avoid the bug.